app/src/main/java/com/github/ybecker/epforuml/latex/MathView.kt

```diff
@@ -0,0 +1,187 @@
+    private var enable_zoom_in_controls = false
+
+    constructor(context: Context) : super(context) {
+        configurationSettingWebView(enable_zoom_in_controls)
+        setDefaultTextColor(context)
+        setDefaultTextSize()
+    }
+
+    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
+        configurationSettingWebView(enable_zoom_in_controls)
+        val mTypeArray = context.theme.obtainStyledAttributes(
+            attrs,
+            R.styleable.MathView,
+            0, 0
+        )
+        try {
+            setBackgroundColor(
+                mTypeArray.getInteger(
+                    R.styleable.MathView_setViewBackgroundColor,
+                    ContextCompat.getColor(context, com.firebase.ui.auth.R.color.fui_transparent)
+                )
+            )
+            setTextColor(
+                mTypeArray.getColor(
+                    R.styleable.MathView_setTextColor,
+                    ContextCompat.getColor(context, R.color.black)
+                )
+            )
+            pixelSizeConversion(
+                mTypeArray.getDimension(
+                    R.styleable.MathView_setTextSize,
+                    default_text_size
+                )
+            )
+            setDisplayText(mTypeArray.getString(R.styleable.MathView_setText))
+            isClickable = mTypeArray.getBoolean(R.styleable.MathView_setClickable, false)
+        } catch (e: Exception) {
+            Log.d(TAG, "Exception:$e")
+        }
+    }
+
+    fun setViewBackgroundColor(color: Int) {
+        setBackgroundColor(color)
+        this.invalidate()
+    }
+
+    private fun pixelSizeConversion(dimension: Float) {
+        if (dimension == default_text_size) {
+            setTextSize(default_text_size.toInt())
+        } else {
+            val pixel_dimen_equivalent_size = (dimension.toDouble() / 1.6).toInt()
+            setTextSize(pixel_dimen_equivalent_size)
+        }
+    }
+
+    @SuppressLint("SetJavaScriptEnabled", "NewApi")
+    private fun configurationSettingWebView(enable_zoom_in_controls: Boolean) {
+        setLayerType(LAYER_TYPE_HARDWARE, null)
+        val settings = this.settings
+        settings.javaScriptEnabled = true
+        settings.allowFileAccess = true
+        settings.displayZoomControls = enable_zoom_in_controls
+        settings.builtInZoomControls = enable_zoom_in_controls
+        settings.setSupportZoom(enable_zoom_in_controls)
+        this.isVerticalScrollBarEnabled = enable_zoom_in_controls
+        this.isHorizontalScrollBarEnabled = enable_zoom_in_controls
+        Log.d(TAG, "Zoom in controls:$enable_zoom_in_controls")
+    }
+
+    fun setDisplayText(formula_text: String?) {
+        display_text = formula_text
+        loadData()
+    }
+
+    fun getDisplayText(): String? {
+        return display_text
+    }
+
+    private val offlineKatexConfig: String
+        private get() {
+            val offline_config = """<!DOCTYPE html>
```

Comment:
again, this is quite harsh to read, could be moved as a const in an accessory class